internal/http/services/owncloud/ocdav/ocdav_blackbox_test.go

```diff
@@ -120,13 +134,383 @@ var _ = Describe("ocdav", func() {
 			},
 		}, nil)
 	})
+	AfterEach(func() {
+		dataSvr.Close()
+	})
 
 	Describe("NewHandler", func() {
 		It("returns a handler", func() {
 			Expect(handler).ToNot(BeNil())
 		})
 	})
 
+	// TODO for every endpoint test the different WebDAV Methods
+
+	// basic metadata
+	// PROPFIND
+	// MKCOL
+	// DELETE
+
+	// basic data
+	// PUT
+	// GET
+	// HEAD
+
+	// move & copy
+	// MOVE
+	// COPY
+
+	// additional methods
+	// PROPPATCH
+	// LOCK
+	// UNLOCK
+	// REPORT
+	// POST (Tus)
+	// OPTIONS?
+
+	Context("at the very legacy /webdav endpoint", func() {
+
+		BeforeEach(func() {
+			// set the webdav endpoint to test
+			basePath = "/webdav"
+
+			// path based requests at the /webdav endpoint first look up the storage space
+			client.On("ListStorageSpaces", mock.Anything, mock.MatchedBy(func(req *cs3storageprovider.ListStorageSpacesRequest) bool {
+				p := string(req.Opaque.Map["path"].Value)
+				return p == "/" || strings.HasPrefix(p, "/users")
+			})).Return(&cs3storageprovider.ListStorageSpacesResponse{
+				Status:        status.NewOK(ctx),
+				StorageSpaces: []*cs3storageprovider.StorageSpace{userspace},
+			}, nil)
+		})
+
+		Describe("PROPFIND", func() {
+
+			var rr *httptest.ResponseRecorder
```

Comment:
How about defining those in a `var` block at the `Context` or the `Describe` level so they can be reused?

Reply:
moved